snowfakery/output_streams.py

```diff
@@ -132,20 +133,27 @@ def __init__(self, stream_or_path=None):
             self.owns_stream = True
         elif stream_or_path is None:
             self.owns_stream = False
-            self.stream = None  # sys.stdout
+            self.stream = sys.stdout
         else:  # noqa
             assert False, f"stream_or_path is {stream_or_path}"
 
-    def close(self) -> Optional[Sequence[str]]:
+    def write(self, data):
+        self.stream.write(data)
+
+    def close(self):
         if self.owns_stream:
             self.stream.close()
             return [f"Generated {self.stream.name}"]
 
 
+class FileOutputStream(OutputStream, SmartStream):
+    pass
+
+
 class DebugOutputStream(FileOutputStream):
     def write_single_row(self, tablename: str, row: Dict) -> None:
         values = ", ".join([f"{key}={value}" for key, value in row.items()])
-        print(f"{tablename}({values})", file=self.stream)
+        self.write(f"{tablename}({values})\n")
```

Comment:
Should we use \r\n on Windows?

Reply:
I think Windows tools support \n these days and I'd rather not pollute the world with more bloated files. By coincidence, I posted this to Facebook this weekend:

"I was debugging something recently and I noticed something horrifying which had never risen to my conscious observation before.

Every time we load a web page or send almost any message across the Internet, the message is broken into many lines. Each line ends with two characters: carriage return, line feed. Why two characters instead of one?

Those two characters are used for compatibility with older protocols from the early Internet which used the two characters. These protocols were compatible with older operating systems like DOS, CP/M, VMS.
These operating systems were trying to be compatible with teletype machines which are more or less electrical long-distance typewriters.

It is only a slight exaggeration to say that many billions (probably trillions) of extra, completely useless, characters are sent over the Internet every day to be compatible with glorified typewriters which were standardized in the 1960s."

(But QUIC will fix it eventually.)

Reply:
I'm fine with just using `\n` until we have a concrete reason to do something more complicated.